### PR TITLE
feat(triggers): readarr

### LINKED ISSUE #128: trigger: Readarr
Would it be possible to add Readarr as one of the triggers in the config.yml? I currently put the path to point to my audiobooks folder location and autoscan usually grabs those files, but it sometimes misses them and I have to manually refresh my audiobooks library in Plex.

### PR DESCRIPTION
**Pull request by voltron4lyfe**
Picking up adding Readarr support from @rg9400's [original ](https://github.com/Cloudbox/autoscan/pull/143). (I checked with him on Discord).  Adds basic Readarr support by supporting the Download event. Future PR's will be needed to add support for Delete and Rename events. 

There was discussion on the original a new router architecture but my understanding of the original PR discussion was that this should move forward and we can figure out how to move this (and other *arr hooks) to the new architecture.

Closes #128, closes #143.

### CODE REVIEW

File: triggers/readarr/readarr.go [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+		l.Error().Err(err).Msg("Processor could not process scans")
+		rw.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
+	l.Info().
+		Str("path", scans[0].Folder).
+		Str("event", event.Type).
+		Msg("Scan moved to processor")
```
Comment (m-rots) [inline triggers/readarr/readarr.go:111]: @l3uddz unsure if this logic is sound, but it's inline with how the Lidarr trigger approaches it.

**Review by m-rots — APPROVED** [review]
Looks good to me! As stated in the PR description, we will just move forward with the "old trigger design" as it currently works fine anyway.

@l3uddz I performed the usual "go test ./..." and manual testing with a http client to make sure the router works correctly. I'm not sure if you're using Readarr (I doubt it) but I'd love some real-world testing. Perhaps we can launch a separate Docker tag for a couple of weeks?

**Review by m-rots — APPROVED** [review]

**Comment by m-rots** [comment]
@l3uddz could you have a look at CI? 😅

**Review by l3uddz — APPROVED** [review]

**Review by m-rots — APPROVED** [review]

**Comment by m-rots** [comment]
We've decided to merge this PR into a temporary `readarr` branch to await some community feedback regarding any bugs which might exist. If no issues arise, then we will stabilise the Readarr trigger in #175.

**Pull request merged**